src/libasr/asr_verify.cpp

```diff
@@ -122,6 +122,18 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         current_symtab = nullptr;
     }
 
+    void visit_Select(const Select_t& x) {
+        bool fall_through = false;
+        for( int i = 0; i < x.n_body; i++ ) {
+            ASR::CaseStmt_t* case_stmt_t = ASR::down_cast<ASR::CaseStmt_t>(x.m_body[i]);
+            fall_through = fall_through || case_stmt_t->m_fall_through;
+        }
+        require(fall_through == x.m_enable_fall_through,
+            "Select_t::m_enable_fall_through should be " +
+            std::to_string(x.m_enable_fall_through));
```

Comment:
That's an even stronger requirement, which is fine. It's always good to start strong, and relax later, than the other way round.